client/state/http/actions.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Internal dependencies
+ */
+import { HTTP_REQUEST } from 'state/action-types';
+
+/***
+ * @typedef {Object} RequestDescription
+ * @property {String} url the url to request
+ * @property {String} method the method we should use in the request: GET, POST etc.
+ * @property {Array<Array<String>>} headers array of [ 'key', 'value' ] pairs for the request headers
+ * @property {Object} body data send as body
+ * @property {Boolean} withCredentials save cookie set on request
```

Comment:
I'm 💯 behind the use of standard names and am glad you bring it up. It was my ignorance that made me ask, though I imagine I'm not the only ignorant one on this.